### PR TITLE
[SPARK-30911][CORE][DOC] Add version information to the configuration of Status

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
1.Add version information to the configuration of `Status`.
2.Update the docs of `Status`.
3.By the way supplementary documentation about https://github.com/apache/spark/pull/27847

I sorted out some information show below.

Item name | Since version | JIRA ID | Commit ID | Note
-- | -- | -- | -- | --
spark.appStateStore.asyncTracking.enable | 2.3.0 | SPARK-20653 | 772e4648d95bda3353723337723543c741ea8476#diff-9ab674b7af7b2097f7d28cb6f5fd1e8c |  
spark.ui.liveUpdate.period | 2.3.0 | SPARK-20644 | c7f38e5adb88d43ef60662c5d6ff4e7a95bff580#diff-9ab674b7af7b2097f7d28cb6f5fd1e8c |  
spark.ui.liveUpdate.minFlushPeriod | 2.4.2 | SPARK-27394 | a8a2ba11ac10051423e58920062b50f328b06421#diff-9ab674b7af7b2097f7d28cb6f5fd1e8c |  
spark.ui.retainedJobs | 1.2.0 | SPARK-2321 | 9530316887612dca060a128fca34dd5a6ab2a9a9#diff-1f32bcb61f51133bd0959a4177a066a5 |  
spark.ui.retainedStages | 0.9.0 | None | 112c0a1776bbc866a1026a9579c6f72f293414c4#diff-1f32bcb61f51133bd0959a4177a066a5 | 0.9.0-incubating-SNAPSHOT
spark.ui.retainedTasks | 2.0.1 | SPARK-15083 | 55db26245d69bb02b7d7d5f25029b1a1cd571644#diff-6bdad48cfc34314e89599655442ff210 |  
spark.ui.retainedDeadExecutors | 2.0.0 | SPARK-7729 | 9f4263392e492b5bc0acecec2712438ff9a257b7#diff-a0ba36f9b1f9829bf3c4689b05ab6cf2 |  
spark.ui.dagGraph.retainedRootRDDs | 2.1.0 | SPARK-17171 | cc87280fcd065b01667ca7a59a1a32c7ab757355#diff-3f492c527ea26679d4307041b28455b8 |  
spark.metrics.appStatusSource.enabled | 3.0.0 | SPARK-30060 | 60f20e5ea2000ab8f4a593b5e4217fd5637c5e22#diff-9f796ae06b0272c1f0a012652a5b68d0 |  

### Why are the changes needed?
Supplemental configuration version information.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
Exists UT

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/internal/config/Status.scala [hunk 1]
```diff
@@ -22,43 +22,52 @@ import java.util.concurrent.TimeUnit
 private[spark] object Status {
 
   val ASYNC_TRACKING_ENABLED = ConfigBuilder("spark.appStateStore.asyncTracking.enable")
+    .version("2.3.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Status.scala]: SPARK-20653, commit ID: 772e4648d95bda3353723337723543c741ea8476#diff-9ab674b7af7b2097f7d28cb6f5fd1e8c

File: core/src/main/scala/org/apache/spark/internal/config/Status.scala [hunk 2]
```diff
@@ -22,43 +22,52 @@ import java.util.concurrent.TimeUnit
...
     .booleanConf
     .createWithDefault(true)
 
   val LIVE_ENTITY_UPDATE_PERIOD = ConfigBuilder("spark.ui.liveUpdate.period")
+    .version("2.3.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Status.scala]: SPARK-20644, commit ID: c7f38e5adb88d43ef60662c5d6ff4e7a95bff580#diff-9ab674b7af7b2097f7d28cb6f5fd1e8c

File: core/src/main/scala/org/apache/spark/internal/config/Status.scala [hunk 3]
```diff
@@ -22,43 +22,52 @@ import java.util.concurrent.TimeUnit
...
     .timeConf(TimeUnit.NANOSECONDS)
     .createWithDefaultString("100ms")
 
   val LIVE_ENTITY_UPDATE_MIN_FLUSH_PERIOD = ConfigBuilder("spark.ui.liveUpdate.minFlushPeriod")
     .doc("Minimum time elapsed before stale UI data is flushed. This avoids UI staleness when " +
       "incoming task events are not fired frequently.")
+    .version("2.4.2")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Status.scala]: SPARK-27394, commit ID: a8a2ba11ac10051423e58920062b50f328b06421#diff-9ab674b7af7b2097f7d28cb6f5fd1e8c

File: core/src/main/scala/org/apache/spark/internal/config/Status.scala [hunk 4]
```diff
@@ -22,43 +22,52 @@ import java.util.concurrent.TimeUnit
...
     .timeConf(TimeUnit.NANOSECONDS)
     .createWithDefaultString("1s")
 
   val MAX_RETAINED_JOBS = ConfigBuilder("spark.ui.retainedJobs")
+    .version("1.2.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Status.scala]: SPARK-2321, commit ID: 9530316887612dca060a128fca34dd5a6ab2a9a9#diff-1f32bcb61f51133bd0959a4177a066a5

File: core/src/main/scala/org/apache/spark/internal/config/Status.scala [hunk 5]
```diff
@@ -22,43 +22,52 @@ import java.util.concurrent.TimeUnit
...
     .intConf
     .createWithDefault(1000)
 
   val MAX_RETAINED_STAGES = ConfigBuilder("spark.ui.retainedStages")
+    .version("0.9.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Status.scala]: No JIRA ID, commit ID: 112c0a1776bbc866a1026a9579c6f72f293414c4#diff-1f32bcb61f51133bd0959a4177a066a5

File: core/src/main/scala/org/apache/spark/internal/config/Status.scala [hunk 6]
```diff
@@ -22,43 +22,52 @@ import java.util.concurrent.TimeUnit
...
     .intConf
     .createWithDefault(1000)
 
   val MAX_RETAINED_TASKS_PER_STAGE = ConfigBuilder("spark.ui.retainedTasks")
+    .version("2.0.1")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Status.scala]: SPARK-15083, commit ID: 55db26245d69bb02b7d7d5f25029b1a1cd571644#diff-6bdad48cfc34314e89599655442ff210

File: core/src/main/scala/org/apache/spark/internal/config/Status.scala [hunk 7]
```diff
@@ -22,43 +22,52 @@ import java.util.concurrent.TimeUnit
...
     .intConf
     .createWithDefault(100000)
 
   val MAX_RETAINED_DEAD_EXECUTORS = ConfigBuilder("spark.ui.retainedDeadExecutors")
+    .version("2.0.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Status.scala]: SPARK-7729, commit ID: 9f4263392e492b5bc0acecec2712438ff9a257b7#diff-a0ba36f9b1f9829bf3c4689b05ab6cf2

File: core/src/main/scala/org/apache/spark/internal/config/Status.scala [hunk 8]
```diff
@@ -22,43 +22,52 @@ import java.util.concurrent.TimeUnit
...
     .intConf
     .createWithDefault(100)
 
   val MAX_RETAINED_ROOT_NODES = ConfigBuilder("spark.ui.dagGraph.retainedRootRDDs")
+    .version("2.1.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Status.scala]: SPARK-17171, commit ID: cc87280fcd065b01667ca7a59a1a32c7ab757355#diff-3f492c527ea26679d4307041b28455b8

File: core/src/main/scala/org/apache/spark/internal/config/Status.scala [hunk 9]
```diff
@@ -22,43 +22,52 @@ import java.util.concurrent.TimeUnit
...
     .intConf
     .createWithDefault(Int.MaxValue)
 
   val METRICS_APP_STATUS_SOURCE_ENABLED =
     ConfigBuilder("spark.metrics.appStatusSource.enabled")
       .doc("Whether Dropwizard/Codahale metrics " +
         "will be reported for the status of the running spark app.")
+      .version("3.0.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Status.scala]: SPARK-30060, commit ID: 60f20e5ea2000ab8f4a593b5e4217fd5637c5e22#diff-9f796ae06b0272c1f0a012652a5b68d0

**Comment by HyukjinKwon** [comment]
Let's merge this one with another PR later just to fix them in batch a bit more.

**Comment by beliefer** [comment]
> Let's merge this one with another PR later just to fix them in batch a bit more.

OK

**Comment by beliefer** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
Seems this is the last ticket identified at this moment. I am merging it.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Also merged to branch-3.0.

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.0.

**Pull request closed without merging**